### PR TITLE
Sokol WebGL2 tests

### PR DESCRIPTION
**Pull request by kripken**
As suggested by @floooh in

https://github.com/emscripten-core/emscripten/issues/11771#issuecomment-667373604

This just adds the 3 WebGL2 ones, which seems like an area we have too
little coverage of.

Fails atm, should pass with  #11780 (verified with code before the regression).

### CODE REVIEW

**Comment by kripken** [comment]
Huh, looks like the GL drivers on CI are not good enough at WebGL2 to run those tests... 2/3 fail on firefox and 3/3 on chrome. Locally all 3 tests pass for me on both browsers...

**Comment by kripken** [comment]
I see 2/3 tests fail with swiftshader locally too, but pass without swiftshader, on chrome beta. On unstable, only 1/3 fails.

@kenrussell should I file chromium issues on WebGL2 bugs with swiftshader?

**Comment by floooh** [comment]
@kripken @kenrussell At least one of the problems (WebGL: INVALID_OPERATION: renderbufferStorageMultisample: samples out of range) looks like I'm trying to create a framebuffer object with a too high sample count (4 MSAA samples). 

In WebGL2 this is actually a runtime-limit which ought to be queried via GL_MAX_SAMPLES (but which I don't do at the moment). If SwiftShader doesn't support 4x MSAA for offscreen framebuffers then this would not be a bug in the WebGL2 SwiftShader backend, but in the sample code (that's something I might need to look into for sokol_gfx.h later too, because at the moment I don't expose the max number of MSAA samples as a runtime limit that can be queried).

A quick workaround might be to disable MSAA in these places:

- remove EMSC_ANTIALIAS here:
    - https://github.com/floooh/sokol-samples/blob/f92aec39e65a02b2286e343c275be36ef937f41b/html5/mipmap-emsc.c#L53
    - https://github.com/floooh/sokol-samples/blob/f92aec39e65a02b2286e343c275be36ef937f41b/html5/mrt-emsc.c#L45
    - https://github.com/floooh/sokol-samples/blob/f92aec39e65a02b2286e343c275be36ef937f41b/html5/arraytex-emsc.c#L36

- and hardwire the sample count here to 1:
    - https://github.com/floooh/sokol-samples/blob/f92aec39e65a02b2286e343c275be36ef937f41b/html5/mrt-emsc.c#L65

However from looking over the CI output logs it looks like only the mrt-emsc.c sample is affected by this, and the other problem in the other samples might be unrelated.

**Comment by kripken** [comment]
Thanks @floooh !

With those changes the tests pass for me with swiftshader locally on chrome beta and unstable, but mipmap fails on stable (so maybe a recent fix?). Let's see what CI says...

**Comment by kenrussell** [comment]
> I see 2/3 tests fail with swiftshader locally too, but pass without swiftshader, on chrome beta. On unstable, only 1/3 fails.
> 
> @kenrussell should I file chromium issues on WebGL2 bugs with swiftshader?

Yes, please file them. Note that the only SwiftShader configuration that is expected to pass full WebGL 2.0 conformance is the combination of ANGLE's Vulkan backend + SwiftShader's Vulkan frontend. I don't think this is shipping by default yet, so very likely the current SwiftShader setup (SwiftShader's older OpenGL ES 3.0 frontend) is expected to fail some tests.

Please do reach out though and we'll get these issues triaged. Feel free to file them under the Blink>WebGL category on http://crbug.com/ , and we'll figure out who from the Chromium, ANGLE or SwiftShader teams should look at them.

**Comment by kripken** [comment]
@kenrussell I think it ended up not necessary, as after @floooh 's suggestions (to avoid the code incorrectly not querying `GL_MAX_SAMPLES`) the only failures I see on swiftshader are on stable, but they pass fine on beta and unstable, so it seems they've been fixed already.

We test with chrome beta on CI, so looks like all tests pass now and this PR is good to go I think. (2 of the tests fail on firefox on CI, which is likely due to the GPU drivers there, as I don't see those results locally; I disabled those for now. Side note, looks like there is [some thinking on swiftshader or such in firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=1548395), which would avoid this, but no active work.)

**Comment by kenrussell** [comment]
> @kenrussell I think it ended up not necessary, as after @floooh 's suggestions (to avoid the code incorrectly not querying `GL_MAX_SAMPLES`) the only failures I see on swiftshader are on stable, but they pass fine on beta and unstable, so it seems they've been fixed already.
> 
> We test with chrome beta on CI, so looks like all tests pass now and this PR is good to go I think. (2 of the tests fail on firefox on CI, which is likely due to the GPU drivers there, as I don't see those results locally; I disabled those for now. Side note, looks like there is [some thinking on swiftshader or such in firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=1548395), which would avoid this, but no active work.)

OK, all sounds good!

File: tests/browser/sokol/HandmadeMath.h [hunk 1]
```diff
@@ -0,0 +1,2886 @@
+// https://github.com/floooh/sokol-samples/blob/master/html5/HandmadeMath.h
```
Comment (sbc100) [inline tests/browser/sokol/HandmadeMath.h:1]: Should we put all this in test/third_party/sokol/?

**Pull request merged**